inspec/a2-iam-v2p1-only-integration/controls/role_projects.rb

```diff
@@ -0,0 +1,134 @@
+          name: "Test Project 1"
+  }
+  CUSTOM_PROJECT_2 = {
+          id: CUSTOM_PROJECT_ID_2,
+          name: "Test Project 2"
+  }
+
+  Projects = [ CUSTOM_PROJECT_1, CUSTOM_PROJECT_2 ]
+
+  describe 'get roles' do
+
+    before(:all) do
+      Projects.each do|project|
+        resp = automate_api_request("/apis/iam/v2beta/projects",
+          http_method: 'POST',
+          request_body: project.to_json
+        )
+        expect(resp.http_status).to eq 200
+      end
+ 
+      Roles.each do|role|
+        resp = automate_api_request("/apis/iam/v2beta/roles",
+          http_method: 'POST',
+          request_body: role.to_json
+        )
+        expect(resp.http_status).to eq 200
+      end
+   end
+
+    after(:all) do
+      Roles.each do|role|
+        resp = automate_api_request("/apis/iam/v2beta/roles/#{role[:id]}", http_method: 'DELETE')
+        expect(resp.http_status).to eq 200
+      end
+      Projects.each do|project|
+        resp = automate_api_request("/apis/iam/v2beta/projects/#{project[:id]}", http_method: 'DELETE')
+        expect(resp.http_status).to eq 200
+      end
+    end
+
+    it 'returns 1 filtered role' do
+      resp = automate_api_request(
+        "/apis/iam/v2beta/roles",
+        request_headers: { 'projects': CUSTOM_PROJECT_ID_1 },
+        )
+      expect(resp.http_status).to eq 200
+      expect(resp.parsed_response_body[:roles].length).to eq 1
+      expect(resp.parsed_response_body[:roles].find { |item| item[:id] == CUSTOM_ROLE_ID_1 }).to_not be_nil
+    end
+
+    it 'returns 2 filtered roles' do
+      resp = automate_api_request(
+        "/apis/iam/v2beta/roles",
+        request_headers: { 'projects': CUSTOM_PROJECT_ID_2 },
+        )
+      expect(resp.http_status).to eq 200
+      expect(resp.parsed_response_body[:roles].length).to eq 2
+      # Would be nice to use the include matcher, but child arrays may differ in order.
+      # expect(resp.parsed_response_body[:roles]).to include(CUSTOM_ROLE_2)
+      # Instead, just check IDs:
+      expect(resp.parsed_response_body[:roles].find { |item| item[:id] == CUSTOM_ROLE_ID_2 }).to_not be_nil
+      expect(resp.parsed_response_body[:roles].find { |item| item[:id] == CUSTOM_ROLE_ID_1 }).to_not be_nil
+    end
+
+    it 'returns no filtered roles for unknown project' do
+      resp = automate_api_request(
+        "/apis/iam/v2beta/roles",
+        request_headers: { 'projects': 'unknown-project' },
+        )
+      expect(resp.http_status).to eq 200
+      expect(resp.parsed_response_body[:roles].length).to eq 0
+    end
+
+    it 'returns some roles that are unassigned' do
+      # these are system default roles
+      resp = automate_api_request(
+        "/apis/iam/v2beta/roles",
+        request_headers: { 'projects': '(unassigned)' },
+        )
+      expect(resp.http_status).to eq 200
+      expect(resp.parsed_response_body[:roles]).not_to be_empty
+    end
+
+    it 'returns all roles when unfiltered' do
```

Comment:
I would add a new control for deny behavior. say we have a policy like this:
```
deny policy:
         statement 1: deny all actions on proj1 roles
         statement 2: allow all actions on proj2, proj3 roles
```

what happens on these requests:
1. GET a role with projects `proj1`?
2. LIST roles with projects `proj1, proj2`?
3. LIST roles with no project headers?
4. DELETE a role with projects `proj1`?
5. UPDATE a role with projects `proj1`?
6. CREATE a role with projects `proj1`?